packages/react/src/elements/components/Typography.js

```diff
@@ -2,77 +2,208 @@ import React from "react";
+  );
+  return (
+    <Typography
+      type="sub1"
+      bold={bold}
+      size={size}
+      color={color}
+      disabled={disabled}
+      text={children}
+    />
+  );
 }
 
-function BodyComponent(props) {
-  return <Typography type="body" text={props.children} />;
+export function Sub2({ children, size, bold, color, disabled }) {
+  console.warn(
+    "Component Sub2 is deprecated and will be removed in the next version"
+  );
+  return (
+    <Typography
+      type="sub2"
+      bold={bold}
+      size={size}
+      color={color}
+      disabled={disabled}
+      text={children}
+    />
+  );
 }
 
-function BoldComponent(props) {
-  return <Typography type="bold" text={props.children} />;
+export function Body({ children, size, bold, color, disabled }) {
+  console.warn(
+    "Component Body is deprecated and will be removed in the next version"
+  );
+  return (
+    <Typography
+      type="body"
+      bold={bold}
+      size={size}
+      color={color}
+      disabled={disabled}
+      text={children}
+    />
+  );
 }
 
-function DisabledComponent(props) {
-  return <Typography type="disabled" text={props.children} />;
+export function Bold({ children, size, bold, color, disabled }) {
+  console.warn(
+    "Component Bold is deprecated and will be removed in the next version"
+  );
+  return (
+    <Typography
+      type="bold"
+      bold={bold}
+      size={size}
+      color={color}
+      disabled={disabled}
+      text={children}
+    />
+  );
 }
 
-function CaptionComponent(props) {
-  return <Typography type="caption" text={props.children} />;
+export function Disabled({ children, size, bold, color, disabled }) {
+  console.warn(
+    "Component Disabled is deprecated and will be removed in the next version"
+  );
+  return (
+    <Typography
+      type="disabled"
+      bold={bold}
+      size={size}
+      color={color}
+      disabled={disabled}
+      text={children}
+    />
+  );
+}
+
+export function Caption({ children, size, bold, color, disabled }) {
+  console.warn(
+    "Component Caption is deprecated and will be removed in the next version"
+  );
+  return (
+    <Typography
+      type="caption"
+      bold={bold}
+      size={size}
+      color={color}
+      disabled={disabled}
+      text={children}
+    />
+  );
 }
 
 const docgenInfo = {
   props: {
+    bold: {
+      description: "Whether to render bold text"
+    },
+    color: {
+      description:
+        "A text color. One of: 'hig-white', 'hig-cool-gray-70', 'hig-blue-50', 'hig-green-good', 'hig-yellow-warning', 'hig-red-alert'"
+    },
+    disabled: {
+      description: "Whether to show text as disabled"
+    },
+    size: {
+      description: "One of: 'small', 'medium', 'large'"
+    },
     children: {
       description: "any content"
     }
   }
 };
 
 const propTypes = {
+  /**
+   * Whether to render bold text
+   */
+  bold: PropTypes.bool,
+  /**
+   * A color to apply to the containing text
+   */
+  color: PropTypes.oneOf([
```

Comment:
It would be cleaner to pull this array of color values from HIGVanilla `Typography.AvailableColors`.